lib/metadataTypes/MetadataType.js

```diff
@@ -721,11 +714,17 @@ class MetadataType {
     static async retrieveSOAP(retrieveDir, buObject, requestParams, additionalFields) {
         requestParams = requestParams || {};
         const fields = this.getFieldNamesToRetrieve(additionalFields);
-        const response = await this.client.soap.retrieveBulk(
-            this.definition.type,
-            fields,
-            requestParams
-        );
+        let response;
+        try {
+            response = await this.client.soap.retrieveBulk(
+                this.definition.type,
+                fields,
+                requestParams
+            );
+        } catch (ex) {
+            this._handleSOAPErrors(ex, 'retrieving');
```

Comment:
is this really needed here? I would rather have all these errors bubble up where possible., rather than return empty objects, which has caused errors in the past.

Reply:
happy to capture that rewrite in a dedicated ticket. for now, this is just adding a missing error handler. also, if we do want to equalize it, this method could later throw an error if we had the time to rewrite and test the surrounding logic - which I do not have right now

Reply:
- added ticket #603